pkgs/development/compilers/dmd/bootstrap.nix

```diff
@@ -0,0 +1,9 @@
+{ callPackage }:
+callPackage ./binary.nix {
+  version = "2.090.1";
```

Comment:
Was there a specific reason why the previous version is used for bootstrapping?
I can think of no technical reason but it would make it even easier if the same version is used for bootstrapping.
This way only one version specifier needs to exist.

Reply:
No particular reason. I followed what `rust` did: https://github.com/NixOS/nixpkgs/blob/7f00b33261d65849027245115d19954ea3e67c5b/pkgs/development/compilers/rust/1_42.nix#L22

Reply:
Ok so they have some technical reasons for that, with dmd there is none, they even build the same compiler again in their Continuous Integration: https://github.com/dlang/dmd/blob/master/.circleci/run.sh#L135

dmd is always guaranteed to be buildable by some old version upwards. That includes the latest.

Reply:
It's not just the version specifier. Also the sha256 of the package vs git would have to be updated. I'm thinking it'll be easier if we just update the bootstrap once in a while.

Reply:
Ok